feat: fetch sensor from sensor_sim + send to pathfinding

JackSMunn opened the pull request:
This pull request includes several changes to the `app` module to enhance the pathfinding service by integrating sensor data and improving the test suite. The most important changes include adding a new schema for sensor data, updating the pathfinding service to query and validate sensor data, and refactoring the test suite to use `pytest` and cover additional scenarios.

### Enhancements to Pathfinding Service:

* **New Sensor Data Schema**:
  - Added `sensor_data_type` schema in `app/schemas/sensor_response_schema.py` to define the structure of sensor data.

* **Integration of Sensor Data**:
  - Updated `app/services/pathfinding_service.py` to query sensor data from the sensor simulation service and include it in the payload for the pathfinding service. [[1]](diffhunk://#diff-0c9078a6d80f739c2785d0db0e8e2d9ca1cb6da6425195b30e171b737420d0faR5-R6) [[2]](diffhunk://#diff-0c9078a6d80f739c2785d0db0e8e2d9ca1cb6da6425195b30e171b737420d0faL23-R29) [[3]](diffhunk://#diff-0c9078a6d80f739c2785d0db0e8e2d9ca1cb6da6425195b30e171b737420d0faR46-L54)

### Refactoring and Improvements:

* **Test Suite Refactoring**:
  - Refactored `app/test/test_api_routes.py` to use `pytest` and added tests for various failure scenarios and successful flow in the `calculate_fastest_path` function.

* **Code Cleanup**:
  - Removed unnecessary `sys.path.append` statement from `app/main.py` to clean up the codebase.- changed main.py to not include the sys.path append that we thought we needed to make module work
- added a sensor response schema
- changed pathfinding_service.py to include sensors in the general pipeline from sensorsim to pathfinding
- changed tests accordingly, now using asyncio

JackSMunn committed:
feat: fetch sensor from sensor_sim + sendoff

- changed main.py to not include the sys.path append that we thought we needed to make module work
- added a sensor response schema
- changed pathfinding_service.py to include sensors in the general pipeline from sensorsim to pathfinding
- changed tests accordingly, now using asyncio

JackSMunn committed:
fix: linting

- removed unused imports

JackSMunn commented on app/test/test_api_routes.py:
```diff
@@ -1,122 +1,179 @@
+VALID_FASTEST_PATH_RESPONSE = {"fastest_path": ["RoomA", "room1", "RoomB"], "distance": 10}
+
+# Helper to run async functions.
+def run_async(coro):
```
running async functions in a sync env

MortenMunk replied:
Why not just use @pytest.mark.asyncio like the other tests? this seem redundant

Jatewo replied:
why not use @pytest.mark.asyncio?

I don't believe it is being called either

Mafusn reviewed (approved):
LGTM 🥬

Jatewo commented on app/main.py:
```diff
@@ -2,12 +2,8 @@
-import os
 import uvicorn
 
-sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
No

Jatewo commented on app/test/test_api_routes.py:
Why not use monkeypatch in a fixture and then pass that fixture to each test? This should work for mocking the env

JackSMunn committed:
update: removed redundant function in test

- changed test_api_routes.py to not include redundant run_async function

Jatewo commented on app/test/test_api_routes.py:
```diff
@@ -1,122 +1,176 @@
+from app.services import pathfinding_service
+
+# Dummy valid data for each service call.
+VALID_ROOM_DATA = {"rooms": [{"id": "room1"}, {"id": "room2"}]}
```
These can be made as pytest fixtures

JackSMunn committed:
update: test_api_routes to accomodate PR requests

- changed the mocking of env to use a fixture using monkeypatch
- changed the two tests relying on it

MortenMunk reviewed (approved):
lgtm

Jatewo reviewed (approved):
lgtm
